### ISSUE TITLE
Running `bundle exec rake` causes Parser warnings on main

### ISSUE THREAD

**Issue by corsonknowles** [rubocop/rubocop#13464]
```
bundle exec rake
Files:         697
Modules:        93 (   13 undocumented)
Classes:       658 (    4 undocumented)
Constants:    1214 ( 1198 undocumented)
Attributes:     38 (    0 undocumented)
Methods:      1530 ( 1335 undocumented)
 27.82% documented
.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.6.0/lib/parser/builders/default.rb:2266: warning: regular expression has redundant nested repeat operator '+': /(?:x+)+/
.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.6.0/lib/parser/builders/default.rb:2266: warning: nested repeat operator '+' and '?' was replaced with '*' in regular expression: /(?:x+)?/
.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.6.0/lib/parser/builders/default.rb:2266: warning: regular expression has ']' without escape: /abc]123/
.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/parser-3.3.6.0/lib/parser/builders/default.rb:2266: warning: regular expression has ']' without escape: /abc]123/
```

--------

## Expected behavior

No  warnings

## Actual behavior

Warning as above

## Steps to reproduce the problem

run 
```bash
bundle exec rake
```

## RuboCop version

```shell
rubocop -V
1.68.0 (using Parser 3.3.6.0, rubocop-ast 1.35.0, analyzing as Ruby 2.7, running on ruby 3.3.5) [arm64-darwin23]
  - rubocop-performance 1.22.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 3.2.0
```

**Comment by dvandersluis** [comment]
I believe the warnings are coming from the parser gem, and there's an open PR: https://github.com/whitequark/parser/pull/1013

**Comment by corsonknowles** [comment]
Thank you so much. I will close this then, as the resolution here should be as simple as pulling in the upstream fix.
